src/HTTPure/Version.purs

```diff
@@ -0,0 +1,41 @@
+module HTTPure.Version
+  ( Version(..)
+  , read
+  ) where
+
+import Prelude
+
+import Node.HTTP as HTTP
+
+-- | These are the HTTP versions that HTTPure understands. There are five
+-- | commonly known versions which are explicitly named.
+data Version
+  = HTTP0_9
+  | HTTP1_0
+  | HTTP1_1
+  | HTTP2_0
+  | HTTP3_0
+  | Other String
+
+-- | If two `Versions` are the same constructor, they are equal.
+derive instance eqVersion :: Eq Version
+
+-- | Allow a `Version` to be represented as a string. This string is formatted
+-- | as it would be in an HTTP request/response.
```

Comment:
I was unsure what the convention was for `Show` instances here. The instance for `Headers` seems to be about displaying it like you'd want for actual HTTP headers (instead of PureScript code). More than happy to change it to whatever.

Reply:
My original plan was to have `Show` instances follow HTTP convention over PureScript convention.  I don't really particularly feel strongly about this, but it seemed to make sense at the time.  I'd prefer to opt for consistency for now and if we decide to double back to the PureScript convention instead we can do it wholesale.